src/types-and-traits.rst

```diff
@@ -2749,3 +2758,67 @@ its lifetime elided form is
 
    fn f <T: ToCStr>(&mut self, args: &[T]) -> &mut Command;
 
+.. _fls_XTBOhK2Yk4lA:
+
+Trait Object Lifetime Elision
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+.. rubric:: Legality Rules
+
+:dp:`fls_URl9CeIVsiWs`
+The :t:`lifetime` of a :t:`trait object type` when :t:`elided` shall be inferred
+as follows:
+
+* :dp:`fls_SHhw6lYHeYyQ`
+  If the :t:`trait object type` is used as the :t:`type specification` of a
+  :t:`reference type`, then the :t:`lifetime` of the :t:`reference type` is the
+  elided :t:`lifetime`,
+
+* :dp:`fls_lC2rwdPLRwaf`
+  If the :t:`trait object type` is used as a :t:`generic argument` and
+
+  * :dp:`fls_e36Hh4oJvfhv`
+    if the corresponding :t:`generic parameter` has exactly one :t:`lifetime`
+    :t:`bound`, then the :t:`lifetime` of that :t:`bound` is the elided
+    :t:`lifetime`,
+
+  * :dp:`fls_ptejalcnIQtm`
+    Otherwise it is a static error to infer the :t:`lifetime` :t:`bound`.
+
+* :dp:`fls_rGbdKtTijby4`
+  If the :t:`trait` of the :t:`trait object type` has exactly one :t:`lifetime`
+  :t:`bound` specified, then the :t:`lifetime` of that :t:`bound` is the
+  inffered :t:`lifetime`,
+
+* :dp:`fls_JhmQpUoExiNZ`
+  If the :t:`trait` of the :t:`trait object type` has noe :t:`lifetime`
+  :t:`[bound]s` specified, then the elided :t:`lifetime` is the ``'static``
+  :t:`lifetime` unless it is elided in :t:`[expression]s` where it instead is
+  inferred,
+
+* :dp:`fls_cglZigwAnASl`
+  Otherwise it is a stic error to infer the :t:`lifetime` :t:`bound`.
+
+
+.. _fls_u5lQkU2rS6uV:
+
+Static Lifetime Elision
+^^^^^^^^^^^^^^^^^^^^^^^
+
+.. rubric:: Legality Rules
+
+:dp:`fls_8irr97rZWfSC`
+The :t:`[lifetime]s` of :t:`[reference type]s` and :t:`paths` in the :t:`type
+specification` of a :t:`constant` or :t:`static` when :t:`elided` shall be
+inferred as the ``'static'`` lifetime.
+
+:dp:`fls_VynJanlhsF8e`
+If :t:`function lifetime elision` for a :t:`lifetime` is applicable, :t:`static
+lifetime elision` is not applied.
```

Comment:
If function lifetime elision is applicable for a lifetime, then static lifetime elision is not applied for that lifetime.